### DISCUSSION TITLE
How do I add customize figures to my KlineChart?

### DISCUSSION THREAD

**Discussion by zbkmr** [klinecharts/KLineChart#433] [Q&A] [answered]
https://klinecharts.com/en-US/guide/figure.html#customize-figure

I'm trying to add new rectext and string to my vue js project using the customize figure feature on the Klinechart website.

1. [// Get the figure class
2. // name is the name of the figure, such as 'arc', 'circle', etc.
3. const Figure = klinecharts.getFigureClass(name)
4. // instantiate and draw
5. // attrs attribute
6. // styles styles
7. // ctx canvas context
8. new Figure(attrs, styles).draw(ctx)](url)

I get this warning when I try the example on the site
vue.esm.js:1906 TypeError: this.kLineChart.getFigureClass is not a function  
Can you share an example for Vue? @liihuu

**Comment by liihuu** [comment]
```javascript
import klinecharts from 'klinecharts'

klinecharts.getFigureClass('line')
```

**Reply by zbkmr** [reply]
@liihuu  first of all thank you for your quick response. I have overcome this problem but now I have another problem.

 ```
   example() {
     const Figure = klinecharts.getFigureClass('line')
     console.log('figure', Figure);
     const attrs= {
      coordinates:{
        x: 1,
        y: 1
      }
     }
     const styles= {
    // style, optional `solid`, `dashed`
    style: 'solid',
    // size
    size: 55,
    // color
    color: '#ce2b2b' ,
    // dotted line parameter
    dashedValue: 25
  }
  const ctx = 'CanvasRenderingContext2D';
     new Figure(attrs, styles).draw(ctx)
    },
```
I am trying to add it to a line with button click, but the line is not added with the information I entered.
I get the error Cannot read properties of undefined (reading 'coordinates').
How can I add the line with the correct example?

**Reply by liihuu** [reply]
e.g.   use line

```vue
<script setup>
import { onMounted } from 'vue'
import klinecharts from 'klinecharts'

onMounted(() => {
  const canvas = document.getElementById('figure_canvas')
  const ctx = canvas.getContext('2d')
  const Figure = klinecharts.getFigureClass('line')
  const attrs = { coordinates: [{ x: 0, y: 0 }, {  x: 150, y: 100  }] }
  const styles = { style: 'dashed', color: '#F99658', size: 4, dashedValue: [6, 6] }
  new Figure(attrs, styles).draw(ctx)
})
</script>

<template>
  <canvas id="figure_canvas" width="200" height="200" />
</template>
```

**Reply by zbkmr** [reply]
<img width="1149" alt="image" src="https://github.com/liihuu/KLineChart/assets/60570361/d2a9d85a-2b8d-47c7-b7bf-05c87f4274c3">

https://codesandbox.io/s/competent-kapitsa-zsj548?file=/src/components/Kline.vue

@liihuu I am getting the same error for your example, the coordinates value cannot be found and it is undefined. What exactly is your Kline version?

**Reply by liihuu** [reply]
Sorry, there are errors in the document,
```javascript
new Figure({ attrs, styles }).draw(ctx)
```

**Reply by zbkmr** [reply]
@liihuu  I didn't understand exactly what you mean. Are you stating that there is an error in your documentation? If this means that I cannot use this feature, should I open a bug ticket?

**Reply by liihuu** [reply, marked as answer]
It can be used, just only docs error.
